### ISSUE TITLE
Clarify whether a different type than 'page', is needed for files other than .mp3 and .mp4

### ISSUE THREAD

**Issue by H-Bluhm** [wellcometrust/workflow#188]
The part of the mets file in question is this:

```
<mets:structMap TYPE="PHYSICAL">
    <mets:div DMDID="DMDPHYS_0000" ID="PHYS_0000" TYPE="physSequence">
      <mets:div CONTENTIDS="urn:nbn:de:gbv:32-1-10000176185" ID="PHYS_0001" ORDER="1" ORDERLABEL=" - " TYPE="page">
        <mets:fptr FILEID="FILE_0001_PRESENTATION"/>
        <mets:fptr FILEID="FILE_0001_DEFAULT"/>
      </mets:div>
```

**Comment by tomcrane** [comment]
Here's what the DDS currently does with the `TYPE` attribute on a physical file.

It uses this attribute, _AND_ the filename, to come up with an internal, DDS type for the asset.

It is aware of `page` as a possible value and will use this directly (asset is therefore of internal type image), but for everything else it analyses the file extension of the physical file, and then falls back to the physical file `TYPE` attribute if it didn't match the file extension to a known list.

So if the filename is `my-sound.mp3` and the TYPE is `mp3`, it's the file extension that tells the DDS it's an MP3. If the filename is `my-sound.mp3` and the TYPE is `page` or `xxx`, it's the file extension that tells the DDS it's an MP3. If the filename is `mysound.xxx` and the TYPE is `mp3`, it's the TYPE that tells the DDS it's an MP3.

The DDS' known list of asset _file extensions_ (not TYPE values) involved in this check is currently:

.mp4
.mp3
.wav
.pdf
.mpg

So basically it is relying on images being of type `page`, but trusting file extensions before types for everything else, and using the type if it didn't deduce the type from the file extension.

This is allowing it to differentiate between mp4 video and mpg, between wav and mp3, for later transcoding operations. It's pretty defensively written as I think it encountered a lot of variation early on.

**Comment by H-Bluhm** [comment]
I am a bit unclear on the 2nd Example of my-sound.mp3 you provided:
you said above that TYPE=page is used directly, then claim here that this file would be processed as an mp3 because of the extension despite TYPE=page, what would be the case here?

We currently only write "mp3", "mp4" and "page" into that field, would it be a better solution to group files by type (i.e. wav gets a TYPE=mp3) or create new types  (i.e. wav gets TYPE=wav).

**Comment by pollecuttn** [comment]
@tomcrane Could you please answer Hauke's question

**Comment by tomcrane** [comment]
Here's the way the DDS makes an internal `fileType` given the physical file `TYPE` attribute (`currentFileType` below), and the file name `amdFileName`:

```
public static string NormaliseFileType(string amdFileName, string currentFileType)
{
    var fileType = currentFileType;
    var filenameLower = amdFileName.ToLowerInvariant();
    if (filenameLower.EndsWith(".mp4"))
    {
        fileType = "mp4";
    }
    if (filenameLower.EndsWith(".mp3") || filenameLower.EndsWith(".wav"))
    {
        fileType = "mp3";
    }
    if (filenameLower.EndsWith(".pdf"))
    {
        fileType = "pdf";
    }
    if (filenameLower.EndsWith(".mpg"))
    {
        fileType = "transcoded-video";
    }
    return fileType;
}
```

It only makes use of the following values:

```
var fileName = physicalFile.AssetMetadata.GetFileName();
var fileType = WdlPackageUtils.NormaliseFileType(fileName, physicalFile.Type);
switch (fileType)
{
    case "page":
        // image-related stuff
        break;
    case "mp4":
        // video-related stuff
        break;
    case "mp3":
        // audio-related stuff
        break;
    case "pdf":
        // pdf-related stuff
        break;
    case "transcoded-video":
        // slightly different video-related stuff
        break;
}
```

As you can see, this is a bit scrappy and has evolved over time. "wav" has been shoe-horned into the existing mechanism very recently. If the DDS is to stick around a while longer I'd like to revisit this.
I'm open to any suggestion here. My ideal would be to have both `type` and `format` in the METS - e.g.,

type = Audio
format = audio/mp3

(where format is an internet type string)

...and not rely on analysing file extensions. But that would require some rework of this logic. In the above code, a .wav file would come out as the DDS type "mp3" which works internally but is not very pretty (audio was always only MP3 until recently). In the above code, TYPE="wav" would be ignored.

I'd welcome a new scheme for identifying types and formats more precisely, and can modify the DDS code to use it, so whatever you feel is best here I can work with.

If you did introduce something new, I assume I would need the DDS to understand the old model for existing METS (rather than update all the existing METS).

**Comment by H-Bluhm** [comment]
Thank you for the additional information.
We'll discuss this further internally and will get back once we have a suggestion on how to proceed here.

**Comment by H-Bluhm** [comment]
Concerning additional information in the metsfile: we cannot provide the format at this point, but the Mimetype is already present at the  sub elemets of <mets:fileGrp USE="OBJECTS">, (e.g. image/jp2). Alternatively the Premis Block has information on the filetype as a reference to PRONOM (e.g. x-fmt/392 is .JP2) which could serve the same purpose.
As for the Type, the contents of this field can be extended as needed.
